core/src/services/hdfs/backend.rs

```diff
@@ -345,97 +326,16 @@ impl Access for HdfsBackend {
         ))
     }
 
-    async fn delete(&self, path: &str, _: OpDelete) -> Result<RpDelete> {
-        let p = build_rooted_abs_path(&self.root, path);
-
-        let meta = self.client.metadata(&p);
-
-        if let Err(err) = meta {
-            return if err.kind() == io::ErrorKind::NotFound {
-                Ok(RpDelete::default())
-            } else {
-                Err(new_std_io_error(err))
-            };
-        }
-
-        // Safety: Err branch has been checked, it's OK to unwrap.
-        let meta = meta.ok().unwrap();
-
-        let result = if meta.is_dir() {
-            self.client.remove_dir(&p)
-        } else {
-            self.client.remove_file(&p)
-        };
-
-        result.map_err(new_std_io_error)?;
-
-        Ok(RpDelete::default())
+    async fn delete(&self, path: &str, args: OpDelete) -> Result<RpDelete> {
+        self.blocking_delete(path, args)
```

Comment:
Thank you very much for this change. However, it's not advisable to use a blocking API call directly (even though we currently do it this way). The ideal approach is to add an async API in [`hdrs`](https://github.com/Xuanwo/hdrs/) first.